drivers/i2c/i2c_cc23x0.c

```diff
@@ -184,6 +204,13 @@ static int i2c_cc23x0_transfer(const struct device *dev, struct i2c_msg *msgs, u
 			break;
 		}
 	}
+
+	while (I2CControllerBusy(config->base)) {
```

Comment:
Looping on hardware like this isn't usually preferred, but if you have no other option please use a WAIT_FOR() macro.